### PR TITLE
feat: forward basic auth user

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR extracts username from the HTTP authorization header and add it to a custom header, providing a user identity that can be directly used in [the Authorization Policy](https://github.com/envoyproxy/gateway/pull/2652) and by other filters or backends.

```
      basicAuth:
        users:
          name: "users-secret1"
        forwardUsernameHeader: x-basic-auth-user
```

Fix: https://github.com/envoyproxy/gateway/issues/2947
Related: https://github.com/envoyproxy/gateway/issues/2250 https://github.com/envoyproxy/gateway/pull/2652

### CODE REVIEW

**Comment by arkodg** [comment]
thoughts on `userNameToHeader` similar to the existing `claimToHeader` field in jwt ?

File: api/v1alpha1/basic_auth_types.go [hunk 1]
```diff
@@ -24,4 +27,10 @@ type BasicAuth struct {
 	//
 	// Note: The secret must be in the same namespace as the SecurityPolicy.
 	Users gwapiv1b1.SecretObjectReference `json:"users"`
+
+	// The name of the HTTP header that will be used to forward the username to the upstream server.
+	// Please note that the header name is a global setting for all the routes in a Gateway.
```
Comment (zhaohuabing) [inline api/v1alpha1/basic_auth_types.go:32]: `UserNameToHeader` can also be per route, but will need some upstream work.

Comment (arkodg) [reply api/v1alpha1/basic_auth_types.go:32]: imo we should wait until per route support is added, else a per route setting will enable this at the global level

Comment (zhaohuabing) [reply api/v1alpha1/basic_auth_types.go:32]: OK, let's wait for the upstream.

**Pull request closed without merging**